docs/src/references/agent/device-profile.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Configuration Management
+tags: [Reference, Device Profile, Firmware Management, Software Management, Configuration Management]
+sidebar_position: 6
+description: Device profile API proposal
+---
+
+# Device profile
+
+A device profile defines any desired combination of a firmware, software and associated configurations to be installed on a device.
+Device profiles are used to get a fleet of devices into a consistent and homogenous state by having the same set of firmware,
+software and configurations installed on all of them.
+
+The `tedge-agent` handles `device_profile` operations as follows:
+
+* Declares `device_profile` support by sending the capability message to `<root>/<device-topic-id>/cmd/device_profile` topics
+* Subscribes to `<root>/<device-topic-id>/cmd/device_profile/+` to receive `device_profile` commands.
+* The `device_profile` command payload is an ordered list of modules representing any firmware, software and configuration combination.
```

Comment:
I'm not fully sure of what you define as a module?

The term *combination* confuses me.
- Is a module homogeneous: either a firmware or a set of software package or a set of configuration files?
- Or is a module a combination of those?
- Or is a module a single firmware/software/configuration sub-operation?

Reading the specs further, I can see you have the third case in mind.
Nevertheless this sentence is confusing.

Reply:
I used the term `module` as a generic term to refer to a single firmware, software or configuration item. And `profile` is any ordered combination of multiple modules of any kind.

Reply:
Though we already use `module` in the software operation data structure, so I'm not sure if that would introduce some confusion there.

We don't necessary have to give it a name, we could just say:

```
* The `device_profile` command payload is an ordered list of items, where an item is either firmware, software and configuration.
```

Reply:
>  The `device_profile` command payload is an ordered list of items

Definitely clearer.